lib/components/Tabs/TabList.tsx

```diff
@@ -1,37 +1,158 @@
-import clsx from 'clsx';
+import { ArrowLeftIcon, ArrowRightIcon } from '@autoguru/icons';
+import { invariant } from '@autoguru/utilities';
+import type { FunctionComponent } from 'react';
 import * as React from 'react';
-import { Children, FunctionComponent } from 'react';
+import {
+	Children,
+	isValidElement,
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from 'react';
 import flattenChildren from 'react-keyed-flatten-children';
 import { useStyles } from 'react-treat';
 
+import { animate, ownerWindow, useEventCallback } from '../../utils';
 import { Box } from '../Box';
-import { IndexContext } from './context';
-import * as styleRefs from './Tabs.treat';
+import { Button } from '../Button';
+import { Icon } from '../Icon';
+import { TabIndexProvider } from './context';
+import * as styleRefs from './TabList.treat';
 
 interface Props {
 	stretch?: boolean;
+	scrollable?: boolean;
 }
 
 export const TabList: FunctionComponent<Props> = ({
 	children,
 	stretch = false,
+	scrollable = false,
 }) => {
+	invariant(
+		!(stretch && scrollable),
+		'Tabs: `stretch={true}` and `scrollable={true}` cannot be used at the same time.',
+	);
+
 	const styles = useStyles(styleRefs);
 
+	const wrapperRef = useRef<HTMLDivElement>(null);
+	const innerRef = useRef<HTMLDivElement>(null);
+
+	const childs = useMemo(
+		() =>
+			Children.map(flattenChildren(children), (child, index) => {
+				if (!isValidElement(child)) return null;
+
+				return (
+					<TabIndexProvider index={index}>{child}</TabIndexProvider>
+				);
+			}),
+		[children],
+	);
+
+	const [displayScroll, setDisplayScroll] = useState({
+		start: false,
+		end: false,
+	});
+
+	const updateScrollButtonState = useEventCallback(() => {
+		if (scrollable) {
+			const {
+				scrollWidth,
+				clientWidth,
+				scrollLeft,
+			} = wrapperRef.current!;
+			const showStartScroll = scrollLeft > 1;
+			const showEndScroll = scrollLeft < scrollWidth - clientWidth - 1;
+
+			if (
+				showStartScroll !== displayScroll.start ||
+				showEndScroll !== displayScroll.end
+			) {
+				setDisplayScroll({
+					start: showStartScroll,
+					end: showEndScroll,
+				});
+			}
+		}
+	});
+
+	const handleOnScroll = useCallback(() => {
+		updateScrollButtonState();
+	}, []);
+
+	const scrollToItem = (delta: number) => {
+		if (wrapperRef.current) {
+			const scrollValue = wrapperRef.current.scrollLeft + delta;
+			animate(wrapperRef.current, 'scrollLeft', scrollValue, 300);
+		}
+	};
+
+	const handleStartButton = () =>
```

Comment:
Should we take the left/right arrows into consideration here?
See how tab 12 never gets fully displayed.

<img width="632" alt="01" src="https://user-images.githubusercontent.com/1334204/82982278-2bfc0600-a031-11ea-88d3-65e6f3cb13fb.png">
<img width="672" alt="02" src="https://user-images.githubusercontent.com/1334204/82982295-37e7c800-a031-11ea-8f1c-9cf4ed9da537.png">